src/averageAges.js

```diff
@@ -34,7 +42,20 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const female = people.filter(woman => woman.sex === 'f');
+  let finalFemale;
+
+  withChildren
+    ? finalFemale = female
+      .filter(woman => people.some(person => person.mother === woman.name))
+    : finalFemale = female;
```

Comment:
You also can use here only one filter method instead of two like with the menAverageAge